components/scim/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -528,12 +528,18 @@ public Group createGroup(Group group) throws CharonException, DuplicateResourceE
                         roleNameWithDomain = UserCoreUtil
                                 .addDomainToName(UserCoreUtil.removeDomainFromName(originalName), domainName);
                     } else if (originalName.indexOf(CarbonConstants.DOMAIN_SEPARATOR) > 0) {
-                        roleNameWithDomain = originalName;
-                        domainName = originalName.split(UserCoreConstants.DOMAIN_SEPARATOR)[0];
+                        domainName = originalName.split(UserCoreConstants.DOMAIN_SEPARATOR)[0].toLowerCase();
```

Comment:
Otherwise there is a conflict when we rename a group. For instance 
1.create domain/group1
2. create DOMAIN/group2
3. rename group2 --> group1
4. we have two entries DOMAIN/group1 and domain/group1